# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/notification/NotificationParent.cpp
# Commit: 807f1d1ada66
# Full Hash: 807f1d1ada66856a25544239f6955f676a1d011e
# Author: Kagami Sascha Rosylight <saschanaz@outlook.com>
# Date: 2024-11-12 09:32:50
# Regressor Bug: 1928702
# File Overlap Count: 1
# Description:
#   Bug 1928702 - Part 4: Implement ActorDestroy r=asuth
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D227689
# ==============================================================================

diff -r 22abfa77d4fc -r 807f1d1ada66 dom/notification/NotificationParent.cpp
--- a/dom/notification/NotificationParent.cpp	Mon Nov 11 20:34:12 2024 +0000
+++ b/dom/notification/NotificationParent.cpp	Mon Nov 11 20:34:12 2024 +0000
@@ -6,6 +6,7 @@
 
 #include "NotificationParent.h"
 
+#include "NotificationUtils.h"
 #include "mozilla/ipc/Endpoint.h"
 
 namespace mozilla::dom::notification {
@@ -40,4 +41,18 @@
   return NS_OK;
 }
 
+void NotificationParent::ActorDestroy(ActorDestroyReason aWhy) {
+  nsAutoString alertName;
+  GetAlertName(alertName);
+  UnregisterNotification(mPrincipal, mId, alertName, CloseMode::InactiveGlobal);
+}
+
+void NotificationParent::MaybeInitAlertName() {
+  if (!mAlertName.IsEmpty()) {
+    return;
+  }
+
+  ComputeAlertName(mPrincipal, mOptions.tag(), mId, mAlertName);
+}
+
 }  // namespace mozilla::dom::notification